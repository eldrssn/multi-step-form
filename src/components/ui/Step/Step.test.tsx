import { render, screen } from '@testing-library/react';
import { Step } from '.';

describe('title and step is correct', () => {
  test('title matches', () => {
    const { container } = render(
      <Step step={{ number: 1, title: 'Title' }} isActive={false} />
    );
    const title = screen.getByText(/title/i);
    expect(title).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('step matches', () => {
    render(<Step step={{ number: 1, title: 'Title' }} isActive={false} />);
    const steps = screen.getAllByText(/1/i);
    expect(steps).toHaveLength(2);
    const titleStep = screen.getByText(/step/i);
    expect(titleStep).toContainHTML('Step 1');
  });
});

describe('style isActive is correct', () => {
  test('when isActive is true', () => {
    render(<Step step={{ number: 1, title: 'Title' }} isActive={true} />);
    const boxElement = screen.getByText(/title/i).closest('.box_active');
    expect(boxElement).toBeInTheDocument();
  });

  test('when isActive is false', () => {
    render(<Step step={{ number: 1, title: 'Title' }} isActive={false} />);
    const boxElement = screen.getByText(/title/i).closest('.box_active');
    expect(boxElement).toBeNull();
  });
});
