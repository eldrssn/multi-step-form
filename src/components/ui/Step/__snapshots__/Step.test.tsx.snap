// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`title and step is correct title matches 1`] = `
<div>
  <div
    class="box"
  >
    <div
      class="number"
    >
      1
    </div>
    <div
      class="info"
    >
      <p
        class="info_number"
      >
        Step 
        1
      </p>
      <p
        class="info_title"
      >
        Title
      </p>
    </div>
  </div>
</div>
`;
